### ISSUE TITLE
An example of a try/except where pylint does not detect UnboundLocalError (used-before-assignment) 

### ISSUE THREAD

**Issue by markmark206** [PyCQA/pylint#5811]
### Bug description

This is related to (but not exactly duplicated by) what is described in https://github.com/PyCQA/pylint/issues/4913 and https://github.com/PyCQA/pylint/issues/4045 .

Here is an example where a variable is initialized inside a try block, and is used in the except, before it had a chance to be initialized.

```
$ cat use_before_assignment.py
```
```python
def compute() -> int:
    raise ValueError("bad chickens")

def count_chickens():
    try:
        chicken_count = compute()
    except ValueError:
        print(chicken_count)

count_chickens()
```

if i run this code, i get an error -- as i would expect. 

```
$ python use_before_assignment.py
Traceback (most recent call last):
  File "/Users/markgenba/src/genba/use_before_assignment.py", line 6, in count_chickens
    chicken_count = compute()
  File "/Users/markgenba/src/genba/use_before_assignment.py", line 2, in compute
    raise ValueError("bad chickens")
ValueError: bad chickens

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/markgenba/src/genba/use_before_assignment.py", line 10, in <module>
    count_chickens()
  File "/Users/markgenba/src/genba/use_before_assignment.py", line 8, in count_chickens
    print(chicken_count)
UnboundLocalError: local variable 'chicken_count' referenced before assignment
```

however, pylint doesn't see a problem with the code:

```
$ pylint -d missing-module-docstring,missing-function-docstring use_before_assignment.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Command used

```shell
pylint -d missing-module-docstring,missing-function-docstring use_before_assignment.py
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

I would expect pylint to report `used-before-assignment` in the line in the except block where we use the variable that was never initialized: 

```python
        print(chicken_count)
```

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.9 (v3.9.9:ccb0e6a345, Nov 15 2021, 13:06:05)
[Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

Mac OS Monteray 12.2 (21D49)

**Comment by markmark206** [comment]
it looks like this issue has already been fixed: https://github.com/PyCQA/pylint/issues/4045#issuecomment-1040809846 closing. thank you!!
